crates/uv-python/src/downloads.rs

```diff
@@ -415,6 +415,72 @@ impl PythonDownloadRequest {
         self.version.take()
     }
 
+    /// Remove default implementation and platform details so the request only contains
+    /// explicitly user-specified segments.
+    #[must_use]
+    pub fn unset_defaults(self) -> Self {
+        self.unset_defaults_with_host(Platform::from_env().ok().as_ref())
+    }
+
+    pub(crate) fn unset_defaults_with_host(mut self, host: Option<&Platform>) -> Self {
+        self.implementation = self
+            .implementation
+            .filter(|implementation_name| *implementation_name != ImplementationName::CPython);
+
+        self.arch = self
+            .arch
+            .filter(|arch| !matches!(arch, ArchRequest::Environment(_)));
+
+        if let Some(host) = host {
+            self.os = self.os.filter(|os| *os != host.os);
+
+            self.libc = self.libc.filter(|libc| *libc != host.libc);
+
+            self.arch = self
+                .arch
+                .filter(|arch| !matches!(arch, ArchRequest::Explicit(explicit_arch) if *explicit_arch == host.arch));
+        }
+
+        self
+    }
+
+    /// Drop patch and prerelease information so the request can be re-used for upgrades.
+    #[must_use]
+    pub fn without_patch(mut self) -> Self {
+        self.version = self.version.take().map(VersionRequest::only_minor);
+        self.prereleases = None;
+        self.build = None;
+        self
+    }
+
+    /// Return a compact string representation suitable for user-facing display.
+    ///
+    /// The resulting string only includes explicitly-set pieces of the request.
```

Comment:
This should explain when it would return `None`, e.g., "Returns [`None`] when no segments are explicitly set"